src/Stateless/TriggerBehaviour.cs

```diff
@@ -9,38 +9,42 @@ public partial class StateMachine<TState, TTrigger>
     {
         internal abstract class TriggerBehaviour
         {
-            readonly TransitionGuard _guard;
+            readonly TransitionGuard _guard;        // null if there is no guard function
```

Comment:
Perhaps, since `TransitionGuard.Conditions` now can be empty, we could default this to a `TransitionGuard` with no conditions, and then delegate some of the functionality below `Guards`, `GuardConditionsMet`, `UnmetGuardConditions` to it so that we don't need null checks in this class?